ktor-network/common/test/io/ktor/network/sockets/tests/UDPSocketTest.kt

```diff
@@ -140,89 +128,124 @@ class UDPSocketTest : CoroutineScope {
 
         socket.socketContext.join()
         assertTrue(socket.isClosed)
-        assertEquals(1, done)
+        assertEquals(1, done.value)
     }
 
     @Test
-    fun testOutgoingInvokeOnClose() = runBlocking {
+    fun testOutgoingInvokeOnClose() = testUdpSockets { selector ->
         val socket: BoundDatagramSocket = aSocket(selector)
             .udp()
             .bind()
 
-        var done = 0
         socket.outgoing.invokeOnClose {
             done += 1
             assertTrue(it is AssertionError)
         }
 
         socket.outgoing.close(AssertionError())
 
-        assertEquals(1, done)
+        assertEquals(1, done.value)
         socket.socketContext.join()
         assertTrue(socket.isClosed)
     }
 
     @Test
-    fun testOutgoingInvokeOnCloseWithSocketClose() = runBlocking {
+    fun testOutgoingInvokeOnCloseWithSocketClose() = testUdpSockets { selector ->
         val socket: BoundDatagramSocket = aSocket(selector)
             .udp()
             .bind()
 
-        var done = 0
         socket.outgoing.invokeOnClose {
             done += 1
         }
 
         socket.close()
 
-        assertEquals(1, done)
+        assertEquals(1, done.value)
 
         socket.socketContext.join()
         assertTrue(socket.isClosed)
     }
 
     @Test
-    fun testOutgoingInvokeOnClosed() = runBlocking {
+    fun testOutgoingInvokeOnClosed() = testUdpSockets { selector ->
         val socket: BoundDatagramSocket = aSocket(selector)
             .udp()
             .bind()
 
         socket.outgoing.close(AssertionError())
 
-        var done = 0
         socket.outgoing.invokeOnClose {
             done += 1
             assertTrue(it is AssertionError)
         }
 
-        assertEquals(1, done)
+        assertEquals(1, done.value)
 
         socket.socketContext.join()
         assertTrue(socket.isClosed)
     }
 
     @Test
     fun testBind() {
-        val socketBuilder: UDPSocketBuilder = aSocket(ActorSelectorManager(Dispatchers.IO)).udp()
+        val socketBuilder: UDPSocketBuilder = aSocket(SelectorManager()).udp()
         val socket = socketBuilder.bind()
         val port = socket.localAddress.port
         socket.close()
 
         repeat(1024) {
             try {
                 socketBuilder
-                    .bind(InetSocketAddress("0.0.0.0", port))
+                    .bind(NetworkAddress("0.0.0.0", port))
                     .close()
-            } catch (_: BindException) {
+            } catch (_: Exception /* TODO: BindException */) {
```

Comment:
I do not know what this test exactly is testing. However, `BindException` is JVM only so I changed it to `Exception` for now. When I ran this test on Kotlin/Native it did not throw any exceptions.

Reply:
It ignores if the specific the `BindException` we should ignore it only for JVM, ignoring `Exception` can hide test failure

Reply:
Fixed: https://github.com/ktorio/ktor/pull/2129/commits/e0e4dfbda0c7113df67771d51c8ddfa94c8a645e